linkml-schema/kbmodel.yaml

```diff
@@ -0,0 +1,141 @@
+id: https://identifiers.org/brain-bican/kb-model
+name: kb-model
+prefixes:
+  linkml: https://w3id.org/linkml/
+  biolink: https://w3id.org/biolink/vocab/
+  bican: https://identifiers.org/brain-bican/vocab/
+  spdx: http://spdx.org/rdf/terms#
+  schema: http://schema.org/
+  ncbi: https://www.ncbi.nlm.nih.gov/Taxonomy/Browser/wwwtax.cgi
+imports:
+  - linkml:types
+  - https://raw.githubusercontent.com/biolink/biolink-model/master/biolink-model
+default_range: string
+default_prefix: bican
+
+classes:
+  gene annotation:
+    is_a: gene
+    description: >-
+      An annotation describing the location, boundaries, and functions of 
+      individual genes within a genome annotation.
+    slots:
+      - referenced in
+      - molecular type
+    id_prefixes:
+      - ENSEMBL
+      - MGI
+      - NCBIGene
+
+  genome annotation:
+    is_a: genome
+    description: >-
+      Location and nomenclature of genes and all of the coding regions in a genome assembly 
+      and the classification of genes and transcripts into types.
+    slots:
+      - reference assembly
+      - version
+      - digest
+      - content_url
+
+  genome assembly:
+    description: >-
+      Genome assembly to contain version and label information
+    attributes:
+      id:
+      taxon:
```

Comment:
perhaps use the same mixin used by gene, such that `in_taxon` is a common reference?

Reply:
i tried to add mixins using `thing with taxon` as well as using `named thing` and `biological entity`, as well as directly quoting the python class syntax as seen in tutorial, but all raise the same error of `genome_assemblies -> 0 -> in_taxon
  extra fields not permitted (type=value_error.extra)` and i also tried directly modifying python classes generated to allow extra fields by `extra = Extra.allow` or ignore or direct strings of those enums, but they raise the same error again. so currently it is still just a separate class of separate fields. I'll continue looking into a possible way